src/components/routes/user-profile/filterEvents.js

```diff
@@ -0,0 +1,3 @@
+export const filterEvents = (events, username) => {
+  return events.filter(event => event.organiser.username === username);
```

Comment:
this could go to the `functions` directory. And we could destructure as well, `({ organiser: {username: organiserUsername} }) => organiserUsername === username`